src/memory/new_delete.cpp

```diff
@@ -28,6 +28,12 @@ void *newImpl(const std::size_t size) {
     return ptr;
   }
 
+  [[maybe_unused]] auto blocker = memgraph::utils::MemoryTracker::OutOfMemoryExceptionBlocker{};
+  auto maybe_msg = memgraph::utils::MemoryErrorStatus().msg();
+  if (maybe_msg) {
+    throw memgraph::utils::OutOfMemoryException{*maybe_msg};
```

Comment:
Does it make sense to do std::move on maybe_msg and update OOM class to accept rvalue ref and move to BasicException